logstash-core/lib/logstash/agent.rb

```diff
@@ -262,4 +268,8 @@ def clean_state?
   def reset_collector
     LogStash::Instrument::Collector.instance.clear
   end
+
+  def setting(key)
```

Comment:
I see who ways to access this settings, one when is from the ivar (using this function) and another when is not  using the ivar. This two methods looks confusing end of the day. what do you think about having them more "uniform"?